quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -132,14 +132,13 @@ ConfigValue getConfigValue(String name, ConfigSourceInterceptorContext context)
 
         Optional<String> configValue = ofNullable(config.getValue());
 
-        if (config.getName().equals(name)) {
```

Comment:
@vmuzikar this check is reponsible for the inconsistency. In the case that the config is explicitly set, then the non-transformed value will be returned.  So in this case for kc.proxy
--proxy=none will return none
omitted it will return false

In removing / moving this check it highlights that this inconistency exists in other places - the isDevModeDatabase check, the DefaultHostnameProvider check of whether http is enabled (it worked before only because the default is false), and when calling the Config.getRawValue - that last one could be a SmallRye issue as raw in that context does not appear to actually refer to the raw config value - https://github.com/smallrye/smallrye-config/blob/5a96de2849c87d144586df07dd4529e28d7e3b85/implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java#L370

There may be some other places as well - so the question is how much should this cleanup be persued or should using the kc. config name always return the untransformed value instead - which seems to have been the original expectation in most places.

Reply:
Thanks for looking into this, @shawkins!

From my perspective, this was a plain oversight that transformers are NOT applied when a config option is explicitly specified by user. @pedroigor Could you please confirm this behaviour is not intentional for some reason?

Luckily, this affects "only" the cases when we fetch the Keycloak config options directly, does not affect the cases when a Keycloak option is mapped to a Quarkus option which is then fetched by some Quarkus code (e.g. `quarkus.http.proxy.proxy-address-forwarding` in case of `kc.proxy`).

With regards to the raw values, feels like a SmallRye bug to me. Should we file an issue for it?

> should using the kc. config name return the untransformed value instead

I don't think so. I don't see a reason why we should transform values only if mapped to some other Quarkus option. Feels counterintuitive. We should leverage the raw values instead.

> Or should we directly use the quarkus property keys where applicable, which would be unambiguous?

I don't think so either. Feels like workaround to me. And not all options map to a Quarkus option, transformers should work for those too.

Reply:
@shawkins It is indeed a very sensitive code path and something we should improve. It is pretty much what @vmuzikar stated and I can confirm the behavior is expected.

@vmuzikar It is not by luck that it only affects `kc` options. The reason is that unmapped options does not have property mappers in our side bound to them.

> With regards to the raw values, feels like a SmallRye bug to me. Should we file an issue for it?

Before doing it, let's make sure it is really an issue as it is very likely to be related to how we are handling/mapping values in our configuration layer.

> I don't think so. I don't see a reason why we should transform values only if mapped to some other Quarkus option. Feels counterintuitive. We should leverage the raw values instead.

It is not only about transforming values when a property is mapped to a Quarkus option. We also have cases where transformation relies on other `kc` options too.

> I don't think so either. Feels like workaround to me. And not all options map to a Quarkus option, transformers should work for those too.

Yeah, we shouldn't reference Quarkus properties at all in our providers/code but rely on our own options. And I think this is the solution we are missing to improve how we are mapping properties. I tried different approaches (including what is being proposed here) but as you can see it breaks the expected behavior.

Perhaps we can sync on this one and look at it together. It has been a while since I don't touch that area and working with someone else on it will be helpful to find a better/right solution.